### PR TITLE
test: api_container_list_test: add filter test

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

### Ⅰ. Describe what this PR did
add test for container list api.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
added

### Ⅳ. Describe how to verify it
```bash
go test -check.f TestList.*
```

### CODE REVIEW

File: test/api_container_list_test.go [hunk 1]
```diff
@@ -25,33 +30,152 @@ func (suite *APIContainerListSuite) SetUpTest(c *check.C) {
...
+func (suite *APIContainerListSuite) TestListFilterInvalid(c *check.C) {
+	success, _, errResp := getContainerListOK(c, "foo", false)
+	c.Assert(success, check.Equals, false)
+	err := util.PartialEqual(errResp.Message, "invalid character")
 	c.Assert(err, check.IsNil)
-	CheckRespStatus(c, resp, 200)
 
-	var got []types.Container
-	err = request.DecodeBody(&got, resp.Body)
+	success, _, errResp = getContainerListOK(c, "{\"foo\":[\"bar\"]}", false)
```
Comment (allencloud) [inline test/api_container_list_test.go]: This is for API test. When test this in cli, we use key=value.
While in API test, the filter data has been transferred to json.

Just comment to record this.

**Comment by allencloud** [comment]
I think all these test cases are very important for users. While I am wondering if we could come to add the missing demand documentation of `filter` functionality.

If we have no guide documentation of this part. No end users would come to use this feature. It will be a feature not open for users.

Could you help to list the filter demand first and add a documentation for this? @zhuangqh

**Comment by allencloud** [comment]
How about adding this in the swagger.yml? In the description of parameter `filter` in APi endpoint `/containers/json`? @zhuangqh

**Comment by zhuangqh** [comment]
> How about adding this in the swagger.yml? In the description of parameter `filter` in APi endpoint `/containers/json`? @zhuangqh

yep. Good suggestion.👍

**Comment by zhuangqh** [comment]
> How about adding this in the swagger.yml? In the description of parameter `filter` in APi endpoint `/containers/json`? @zhuangqh
> 
> If we did this, the @pouchrobot would automatically generate the corresponding document for us.

@allencloud updated

File: apis/swagger.yml [hunk 2]
```diff
@@ -2356,7 +2358,6 @@ definitions:
             items:
               type: "string"
           OomScoreAdj:
-            type: "integer"
```
Comment (allencloud) [inline apis/swagger.yml]: This is duplicated as LIne 2363. record this.

File: apis/swagger.yml [hunk 3]
```diff
@@ -518,6 +516,10 @@ paths:
           description: "Return all containers. By default, only running containers are shown"
           type: "boolean"
           default: false
+        - name: "filters"
+          in: "query"
+          description: "Filters in json string(type map[string][]string in Golang). Support filter key [ id label name status ]. More detailed, [ id name status ] support regular expression. [label] support equal and unequal operator."
```
Comment (allencloud) [inline apis/swagger.yml]: Could you change this to make it more readable, like : 
https://github.com/moby/moby/blob/master/api/swagger.yaml#L4325-L4341

**Comment by allencloud** [comment]
LGTM

**Pull request merged**